pkg/frontend/config.go

```diff
@@ -79,7 +79,7 @@ func InitFrontend(cfg CombinedFrontendConfig, limits v1.Limits, grpcListenPort i
 			cfg.FrontendV2.Port = grpcListenPort
 		}
 
-		fr, err := v2.NewFrontend(cfg.FrontendV2, log, reg)
+		fr, err := v2.NewFrontend(cfg.FrontendV2, limits, prometheusCodec, log, reg)
```

Comment:
currently only applying this to the v2 frontend - I was initially throw off by a comment from someone saying that the v1 frontend doesn't use the scheduler, but it appears the v1 frontend uses the scheduler code in-process, just not as another component.

Should this be applied to v1 frontend as well?

Reply:
>I was initially throw off by a comment from someone saying that the v1 frontend doesn't use the scheduler

I was that someone - sorry for throwing you off.

>Should this be applied to v1 frontend as well?

How difficult would it be? If it's anything more than trivial, I'm not sure it's worth it - in any substantial cluster, we recommend using the separate query-scheduler (and therefore v2).

(Separately, I wonder if v1 is something we should deprecate and remove, but that's something well beyond the scope of this PR.)

Reply:
I think let's leave v1 as-is for now, that can be a separate PR if we decide we want to support this there. I think we need to clarify if v1 is something we want to continue supporting - would you be able to start a discussion with the team on this?